pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,383 @@
+	existingMachinePoolList       []clustersmgmtv1.MachinePool
+	selectedClusterMachinePool    clustersmgmtv1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	pageNumber                    int
+	pageSize                      int
+
+	f *factory.Factory
+}
+
+// list of consts should come from KFM
+const (
+	machinePoolId           = "kafka-standard"
+	machinePoolTaintKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolTaintEffect  = "NoExecute"
+	machinePoolTaintValue   = "standard"
+	machinePoolInstanceType = "r5.xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+	clusterReadyState       = "ready"
+	fleetshardAddonId       = "kas-fleetshard-operator"
+	strimziAddonId          = "managed-kafka"
+	fleetshardAddonIdQE     = "kas-fleetshard-operator-qe"
+	strimziAddonIdQE        = "managed-kafka-qe"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "register-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.registerCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.registerCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.registerCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.registercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterId.description"))
+	flags.IntVar(&opts.pageNumber, "page-number", int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber)), f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageNumber.description"))
+	flags.IntVar(&opts.pageSize, "page-size", 100, f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageSize.description"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) (err error) {
+	opts.pageNumber = int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber))
+	err = getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	if len(opts.clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.registerCluster.run.noClusterFound")
+	}
+
+	if opts.selectedClusterId == "" {
+		err = runClusterSelectionInteractivePrompt(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range opts.clusterList {
+			cluster := opts.clusterList[i]
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
```

Comment:
This won't work in all the cases:
if the clusterID given via the  opts.selectedClusterId is not in the curent page size.
My suggestion will be to take this onto account when fetching the list of cluster.
i.e in https://github.com/redhat-developer/app-services-cli/blob/bc6732468ce784386222b13bcf44777d4b2e2929/pkg/cmd/dedicated/register/registercluster.go#L120 construct a search query that has `id = 'the selected cluster id'`. This can be built on top of  https://github.com/redhat-developer/app-services-cli/pull/1830/files#r1132713648